basic_hashtable/b_hashtables.c

```diff
@@ -84,7 +89,17 @@ BasicHashTable *create_hash_table(int capacity)
  ****/
 void hash_table_insert(BasicHashTable *ht, char *key, char *value)
 {
-
+  unsigned int index = hash(key, ht->capacity);
+
+  if (ht->storage[index] != NULL)
```

Comment:
instead of
```
if (ht->storage[index] != NULL)
  {
    if (strcmp(ht->storage[index]->key, key) == 0)
```
another way to do it
```
   if (ht->storage[index] && strcmp(ht->storage[index]->key, key) != 0)
```